src/devices/Mcp25xxx/Register/AcceptanceFilter/RxFxEid0.cs

```diff
@@ -0,0 +1,105 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+
+namespace Iot.Device.Mcp25xxx.Register.AcceptanceFilter
+{
+    /// <summary>
+    /// Filter Extended Identifier Low Register.
+    /// </summary>
+    public class RxFxEid0 : IRegister
+    {
+        /// <summary>
+        /// Initializes a new instance of the RxFxEid0 class.
+        /// </summary>
+        /// <param name="rxFilterNumber">Receive Filter Number.  Must be a value of 0 - 5.</param>
+        /// <param name="eid">
+        /// Extended Identifier bits.
+        /// These bits hold the filter bits to be applied to bits[7:0] of the Extended Identifier portion of a received
+        /// message or to Byte 1 in received data if corresponding with RXM[1:0] = 00 and EXIDE = 0.
+        /// </param>
+        public RxFxEid0(byte rxFilterNumber, byte eid)
+        {
+            if (rxFilterNumber > 5)
+            {
+                throw new ArgumentException($"Invalid RX Filter Number value {rxFilterNumber}.", nameof(rxFilterNumber));
+            }
+
+            RxFilterNumber = rxFilterNumber;
+            Eid = eid;
+        }
+
+        /// <summary>
+        /// Receive Filter Number.  Must be a value of 0 - 5.
+        /// </summary>
+        public byte RxFilterNumber { get; }
+
+        /// <summary>
+        /// Extended Identifier bits.
+        /// These bits hold the filter bits to be applied to bits[7:0] of the Extended Identifier portion of a received
+        /// message or to Byte 1 in received data if corresponding with RXM[1:0] = 00 and EXIDE = 0.
+        /// </summary>
+        public byte Eid { get; }
+
+        private Address GetAddress()
+        {
+            switch (RxFilterNumber)
+            {
+                case 0:
+                    return Address.RxF0Eid0;
```

Comment:
optional: I think would smplify a bit if you converted this to array and just simply use indexer and IndexOf (I think Span has it somewhere) for the other method

Reply:
same comment for similar classes (actually wondering if we should unify them into one class)

Reply:
This would be better approach.  I started to update and get an error I can't seem to understand though.

This example works (not related to binding):

```csharp
ReadOnlySpan<byte> value = new byte[] { 0x00, 0x01, 0x02 };
value.IndexOf<byte>(0x00);
```

Below I get the following and hot sure why it is trying to use the overloaded method call:

Error	CS1503	Argument 2: cannot convert from 'Iot.Device.Mcp25xxx.Register.Address' to 'System.ReadOnlySpan<Iot.Device.Mcp25xxx.Register.Address>'

```csharp
ReadOnlySpan<Address> addresses = new Address[] {Address.TxB0Sidl, Address.TxB1Sidl, Address.TxB2Sidl };
addresses.IndexOf<Address>(Address.TxB0Sidl);  // Get red squiggles here.
```

Have any thoughts on this?

Reply:
I think you can workaround by creating `stackalloc Address[1] { whateverYouSearchFor }` since this is what it seems to expect. I would presume searching for single element should work though. cc: @ahsonkhan @GrabYourPitchforks

Reply:
I tried this and works, but is not ideal.

```csharp

Address address = Address.RxB1Dlc;

ReadOnlySpan<byte> addresses = stackalloc byte[2]
{
    (byte)Address.RxB0Dlc,
    (byte)Address.RxB1Dlc
};

int index = addresses.IndexOf((byte)address);

if( index < 0)
{
    throw new ArgumentException($"Invalid address value {address}.", nameof(address));
}

return (byte)index;
```

Reply:
@joperezr do you have any thoughts on how to fix this?  I still see error in newest VS2019 preview, as well.

Reply:
The reason why this happens is because the extension method IndexOf requires that the T type should be IEquatable<T>, and an Enum will derive from System.Enum which is not IEquatable<T>. You could theoretically cast it as a byte if you wanted which should work, or alternatively, instead of having Address be an Enum, you could have it be a struct that does extends byte and has some default static values. For that suggestion, you can take a look at what we did with PinValue here: https://github.com/dotnet/iot/blob/master/src/System.Device.Gpio/System/Device/Gpio/PinValue.cs

Reply:
As an aside, I am not sure if it makes sense to use `IndexOf` to search through a compile-time known enum set like this and the switch/case makes more sense to me. Why incur the cost of searching through the enum array every time? What's the benefit?

Reply:
@ahsonkhan Thanks for feedback.  This change was based on other feedback for the PR.  @krwq If you're good with current code, I believe there are no more outstanding issues with PR.